### ISSUE TITLE
Nexus support zip won't generate

### ISSUE THREAD

**Issue by hgfranco** [juliogonzalez/nexus-oss-rpms#27]
It appears the spec file is generating the log file as nexus3.log.
`https://github.com/juliogonzalez/nexus-oss-rpms/blob/master/SPECS/nexus3-oss.spec#L71`

Because of this Nexus is unable to generate a support zip (in Nexus administration UI under "support/support zip"). It should be called nexus.log.

```
2018-11-06 17:09:00,926+0000 WARN  [qtp1032748576-49]  admin org.sonatype.nexus.internal.log.LogbackLogManager - Log file does not exist: nexus.log

2018-11-06 17:09:00,936+0000 ERROR [qtp1032748576-49]  admin org.sonatype.nexus.internal.atlas.SupportZipGeneratorImpl - Failed to create support ZIP
```

**Comment by rseddon** [comment]
Related issue:  https://issues.sonatype.org/browse/NEXUS-18337

**Comment by juliogonzalez** [comment]
@hgfranco, @rseddon, thanks for the report and the information.

For me this is clearly a Nexus bug. Otherwise the log path should not be configurable.

That said:

@hgfranco: the line 71 not only changes the filename, but the path as well.

By default the path is `/var/log/nexus/nexus.log`, and to keep coherent with the rest of the RPM, this is changed to `/var/log/nexus3/nexus3.log`

Does the support zip work if the filepath is `/var/log/nexus3/nexus.log`?

If you can confirm that, I could create a symlink for next release, if the fix to `NEXUS-18337` takes too long or if it is closed as invalid by the Nexus team.

**Comment by rseddon** [comment]
@juliogonzalez 

Agreed, it's a Nexus bug (and I filed it as such).  Yes, it will work if the file path is /var/log/nexus3/nexus.log.

**Comment by juliogonzalez** [comment]
It seems that 3.15.0-01 is out, but the bug is still present.

I will try to see tomorrow if I can fix the code with a patch to nexus to 3.16 (unlikely, since I am not a Java developer), and otherwise will accept the update from @Angelo-Verona at  #32 (thanks!) and will create the symbolic link until the bug is fixed.

**Comment by Angelo-Verona** [comment]
@juliogonzalez 
I did try to create symbolic link from nexus3/nexus.log -> nexus3.log but it didn't work for me. At least on CentOS 7.
I assume that nexus trying some check if it's a correct file and if it is not a regular file it will not take it.
I decided to rewrite logback to nexus.log

**Comment by juliogonzalez** [comment]
Thanks for confirming @Angelo-Verona.

I had a quick look at Nexus' code, and it seems the filename `nexus.lorg` is hardcoded in several places, including a plugin:
```
components/nexus-base/src/main/java/org/sonatype/nexus/internal/log/LastShutdownTimeServiceImpl.java:51:  private static final String LOG_FILE_NAME = "nexus.log";
components/nexus-base/src/test/java/org/sonatype/nexus/internal/log/LastShutdownTimeServiceImplTest.java:71:    when(logManager.getLogFile("nexus.log")).thenReturn(nexusFile);
components/nexus-base/src/test/java/org/sonatype/nexus/internal/log/LastShutdownTimeServiceImplTest.java:78:    when(logManager.getLogFile("nexus.log")).thenReturn(null);
components/nexus-base/src/test/java/org/sonatype/nexus/internal/log/LastShutdownTimeServiceImplTest.java:88:    when(logManager.getLogFile("nexus.log")).thenReturn(emptyFile);
components/nexus-pax-exam/src/main/java/org/sonatype/nexus/pax/exam/NexusPaxExamSupport.java:552:    testIndex.recordAndCopyLink("nexus.log", new File(logDir, "nexus.log"));
plugins/nexus-coreui-plugin/src/main/java/org/sonatype/nexus/coreui/internal/log/LogResource.java:82:    InputStream log = logManager.getLogFileStream("nexus.log", from, count);
```

I tried to see how to retrieve the value from `logback.xml` but can't find it.

So I am afraid this exceeds my (really basic) knowledge of Java, not to mention the absolute lack of knowledge of Nexus source code, to make such changes.

Honestly, I don't think Sonatype will fix this soon, unless there's a community contribution, so I am wondering if the RPM should stop nexus, move nexus3.log to nexus.log, change logback.xml, create a link from nexus3.log to nexus.log, give a warning to the administrator and start nexus again.

What's your opinion?

**Comment by Angelo-Verona** [comment]
it is in`/etc/nexus3/logback/logback.xml` the line 
```
....
  <appender name="logfile" class="ch.qos.logback.core.rolling.RollingFileAppender">
    <File>/var/log/nexus3/nexus3.log</File>
....
```
fastest route to fix it is probably in RPM specfile as %post script
`sed -i 's#<File>/var/log/nexus3/nexus3.log</File>#<File>/var/log/nexus3/nexus.log</File>#' /etc/nexus3/logback/logback.xml`

I think that we can manipulate with Nexus, I am usually against changes in the source but RPM is about adjusting the application to the system, and this is bummer if its fail after install. If they fix that we can simply remove the workaround. The worst thing is that you will end up with two logs on the end which is not a breakdeal.

**Comment by juliogonzalez** [comment]
Yes, I know. In fact I was am changing `logback.xml` at the spec [during installation](https://github.com/juliogonzalez/nexus-oss-rpms/blob/master/SPECS/nexus3-oss.spec#L71), to match the package name and service name (and that's the reason the supportfile generation fails, as it seems the name is hardcoded).

I will approve your PR for the new version, and will implement a workaround soon, copying the log, linking and warning the user. It should be simple.

**Comment by juliogonzalez** [comment]
Regarding this issue, I did not solve it at the spec yet, but at least if you now modify the config files, the changes should stay when updating the package (this change comes with the last SPECs for Nexus 2 and 3 that I just released)

**Comment by juliogonzalez** [comment]
Fixed on 3.17.0. The SPEC file is already released (working on 3.18.X now)
